--- a/payloads/external/edk2/workspace/mrchromebox/UefiPayloadPkg/UefiPayloadPkg.dsc
+++ b/payloads/external/edk2/workspace/mrchromebox/UefiPayloadPkg/UefiPayloadPkg.dsc
@@ -180,6 +180,7 @@
   DEFINE NETWORK_RTEK_PCI               = FALSE
   DEFINE NETWORK_RTEK_USB               = FALSE
   DEFINE NETWORK_ASIX_USB3              = FALSE
   DEFINE NETWORK_ASIX_USB2              = FALSE
+  DEFINE NETWORK_RTEK_MAC_FROM_CBFS     = TRUE
 
 !if $(NETWORK_PXE_BOOT) == TRUE
   DEFINE NETWORK_SNP_ENABLE             = TRUE
