--- a/src/mainboard/google/puff/Kconfig
+++ b/src/mainboard/google/puff/Kconfig
@@ -32,6 +32,8 @@
 	select RT8168_GEN_ACPI_POWER_RESOURCE
 	select RT8168_GET_MAC_FROM_VPD
 	select RT8168_SET_LED_MODE
+	select RT8168_PUT_MAC_TO_ERI
+	select REALTEK_8168_RESET
 	select SOC_INTEL_COMETLAKE_1
 	select SOC_INTEL_CSE_LITE_SKU
 	select SPD_CACHE_IN_FMAP
@@ -76,6 +78,7 @@
 config BOARD_GOOGLE_PUFF
 	select BOARD_GOOGLE_BASEBOARD_PUFF
 	select BOARD_ROMSIZE_KB_32768
+	select RT8168_PUT_MAC_TO_ERI
 
 config BOARD_GOOGLE_SCOUT
 	select BOARD_GOOGLE_BASEBOARD_PUFF