--- a/src/drivers/net/r8168.c
+++ b/src/drivers/net/r8168.c
@@ -125,6 +125,7 @@ static enum cb_err fetch_mac_vpd_dev_idx(u8 *macstrbuf, u8 device_index)
 	char key[] = "ethernet_mac "; /* Leave a space at tail to stuff an index */
 
 	/*
 	 * Map each NIC on the DUT to "ethernet_macN", where N is [0-9].
 	 * Translate index number from integer to ascii by adding '0' char.
 	 */
 	key[DEVICE_INDEX_BYTE] = device_index + '0';
+	printk(BIOS_DEBUG, "r8168: fetch_mac_vpd_dev_idx: device_index=%d, key='%s'\n", device_index, key);
 
 	return fetch_mac_vpd_key(macstrbuf, key);
 }
