diff --git a/src/mainboard/google/puff/variants/kaisa/overridetree.cb b/src/mainboard/google/puff/variants/kaisa/overridetree.cb
index 0b539a1a0b..41b56ae722 100644
--- a/src/mainboard/google/puff/variants/kaisa/overridetree.cb
+++ b/src/mainboard/google/puff/variants/kaisa/overridetree.cb
@@ -2,8 +2,8 @@ chip soc/intel/cannonlake
 	register "tcc_offset" = "5"	# TCC of 95C
 
 	register "power_limits_config" = "{
-		.tdp_pl1_override = 15,
-		.tdp_pl2_override = 51,
+		.tdp_pl1_override = 25,
+		.tdp_pl2_override = 50,
 	}"
 
 	# Auto-switch between X4 NVMe and X2 NVMe.
@@ -284,18 +284,18 @@ chip soc/intel/cannonlake
 				register "policies.critical[1]" = "DPTF_CRITICAL(TEMP_SENSOR_0, 78, SHUTDOWN)"
 
 				## Power Limits Control
-				# PL1 is fixed at 15W, avg over 28-32s interval
-				# 15-51W PL2 in 1000mW increments, avg over 28-32s interval
+				# PL1 is fixed at 25W, avg over 64s interval
+				# 25-50W PL2 in 1000mW increments, avg over 32s interval
 				register "controls.power_limits.pl1" = "{
-					.min_power = 15000,
-					.max_power = 15000,
-					.time_window_min = 28 * MSECS_PER_SEC,
-					.time_window_max = 32 * MSECS_PER_SEC,
+					.min_power = 25000,
+					.max_power = 25000,
+					.time_window_min = 64 * MSECS_PER_SEC,
+					.time_window_max = 64 * MSECS_PER_SEC,
 					.granularity = 200,}"
 				register "controls.power_limits.pl2" = "{
 					.min_power = 15000,
-					.max_power = 51000,
-					.time_window_min = 28 * MSECS_PER_SEC,
+					.max_power = 50000,
+					.time_window_min = 32 * MSECS_PER_SEC,
 					.time_window_max = 32 * MSECS_PER_SEC,
 					.granularity = 1000,}"
 
diff --git a/src/soc/intel/cannonlake/systemagent.c b/src/soc/intel/cannonlake/systemagent.c
index 5cb3418ed6..c88fbcf045 100644
--- a/src/soc/intel/cannonlake/systemagent.c
+++ b/src/soc/intel/cannonlake/systemagent.c
@@ -61,7 +61,7 @@ void soc_systemagent_init(struct device *dev)
 	mdelay(1);
 	config = config_of_soc();
 	soc_config = &config->power_limits_config;
-	set_power_limits(MOBILE_SKU_PL1_TIME_SEC, soc_config);
+	set_power_limits(64, soc_config); // Changed from MOBILE_SKU_PL1_TIME_SEC (28s) to 64s for high performance
 }
 
 uint32_t soc_systemagent_max_chan_capacity_mib(u8 capid0_a_ddrsz)
